### ISSUE TITLE
Remove falsy dict.get() fallback in `if` statements

### ISSUE THREAD

**Issue by janosh** [astral-sh/ruff#4934]
```py
# bad
if dct.get(key, False):
if dct.get(key, ""):
if dct.get(key, []):
if dct.get(key, {}):

# good
if dct.get(key):

# bad
if dct.get(key, set()) and other_cond:
if dct.get(key, 0) or other_cond:

# good
if dct.get(key) and other_cond:
if dct.get(key) or other_cond:
```

**Comment by evanrittenhouse** [comment]
@charliermarsh Would this just be a `RUF` rule?

**Comment by janosh** [comment]
Actually if statement is not needed. Any boolean evaluation.

**Comment by dhruvmanila** [comment]
```python
# bad
if dct.get(key) and other_cond:
if dct.get(key) or other_cond:
```

Can you expand on why this is bad? I don't think it is as the `other_cond` will be evaluated if the `.get` is truthy in the `and` case and falsy in the `or` case.

This would also require type inference without which we're limited to certain cases.

**Comment by evanrittenhouse** [comment]
@dhruvmanila just a heads up, I don't think the boolean expression is necessary (see the comment above yours :) )

**Comment by janosh** [comment]
> Can you expand on why this is bad?

Sorry, that was an oversight. I meant `# good`. Fixed above.
